lib/azure/armrest/configuration.rb

```diff
@@ -0,0 +1,183 @@
+module Azure
+  module Armrest
+    class Configuration
+      # Used to store unique token information.
+      @@tokens = Hash.new([])
+
+      # The client ID used to gather token information.
+      attr_accessor :client_id
+
+      # The client key used to gather token information.
+      attr_accessor :client_key
+
+      # The tenant ID used to gather token information.
+      attr_accessor :tenant_id
+
+      # The resource group used for a given http request.
+      attr_accessor :resource_group
+
+      # The API version used for a given http request. The default is 2015-01-01.
+      attr_accessor :api_version
+
+      # The grant type. The default is client_credentials
+      attr_accessor :grant_type
+
+      # The content type specified for http requests. The default is 'application/json'
+      attr_accessor :content_type
+
+      # The accept type specified for http request results. The default is 'application/json'
+      attr_accessor :accept
+
+      # Explicitly set the token.
+      attr_writer :token
+
+      # Yields a new Azure::Armrest::Configuration objects. Note that you must
+      # specify a client_id and client_key. All other parameters are optional.
+      #
+      # Example:
+      #
+      #   config = Azure::Armrest::Configuration.new(
+      #     :client_id  => 'xxxx',
+      #     :client_key => 'yyyy',
+      #     :tenant_id  => 'zzzz'
+      #   )
+      #
+      def initialize(hash)
+        @subscriptions = Hash.new([]) # Used internally
```

Comment:
I don't know that I agree with that. I think that is something for end users to implement on their end if they want.

But, I'll continue down that path for now. Since you brought up config equality, I'm thinking this class should implement Comparable.